python/mxnet/model.py

```diff
@@ -916,3 +917,89 @@ def create(symbol, X, y=None, ctx=None,
                   work_load_list=work_load_list,
                   eval_batch_end_callback=eval_batch_end_callback)
         return model
+
+
+
+def _find_nodes_inputs(node_dict, arg_dict):
+    """Find the inputs of a given node in symbol graph
+
+    Parameters
+    ----------
+    node_dict: dict
+        Decoded node json in dict format
+    arg_dict: dict
+        Argument dict, this is taken from symbol json directly
+
+    Returns
+    -------
+    inputs: list
+        A list of input node info
+    """
+    inputs = []
+    for node_idx, output_idx in node_dict["inputs"]:
+        if node_idx not in arg_dict:
+            inputs.append([node_idx, output_idx])
+    return inputs
+
+def _get_feature_symbol(sym_json):
+    """Get feature extractor symbol from the given symbol
+    .. note::
+        This function is NOT working with multiple output network
+
+    Parameters
+    ----------
+    sym_json: str
+        symbol json in string format
+
+    Returns
+    -------
+    sym: mx.symbol.Symbol
+        symbol of feature extractor part
+    """
+
+    sym_dict = json.loads(sym_json)
+    arg_nodes = sym_dict["arg_nodes"]
+    arg_dict = dict(zip(arg_nodes, [True for i in range(len(arg_nodes))]))
+    heads = sym_dict["heads"]
+    nodes = sym_dict["nodes"]
+
+    if len(heads) != 1:
+        raise Exception("Only support single output model")
+    head_idx = heads[0][0]
+    # get linear classifier node of original model
+    classifier = _find_nodes_inputs(nodes[head_idx], arg_dict)
+    if len(classifier) != 1:
+        raise Exception("Only support single output model")
+
+    # get feature node from classifier node
+    feature = _find_nodes_inputs(nodes[classifier[0][0]], arg_dict)
+    if len(feature) != 1:
+        raise Exception("Only support single output model")
+
+    # make new symbol without classifier
+    new_nodes = nodes[:feature[0][0] + 1]
+    new_arg_nodes = [idx for idx in arg_nodes if idx < feature[0][0]]
+    new_sym = {"arg_nodes" : new_arg_nodes, "heads" : feature, "nodes":new_nodes}
+    return sym.load_json(json.dumps(new_sym))
+
+def finetune(model, num_new_output, **kwargs):
+    """Get a FeedForward model for fine-tune
+    .. note::
+        This function is not working with multiple outputs network
+
+    Parameters
+    ----------
+    model: mx.model.FeedForward
+        Model which contains parameters which will be used for fine-tune
+    num_new_output: int
+        New number of output category
+    kwargs: kwargs
+        mx.model.create function's parameter
+    """
+    sym_json = model.symbol.tojson()
+    feature = _get_feature_symbol(sym_json)
+    classifier = sym.FullyConnected(feature, num_hidden=num_new_output)
+    classifier = sym.SoftmaxOutput(classifier, name="softmax")
```

Comment:
Here can we use the same output layer as the original network, i.e. (same type, same name)? This will make finetune work for Regression, BinaryClassification and MulticlassClassification.